tls/ttls.c

```diff
@@ -1670,16 +1670,28 @@ ttls_parse_certificate(TlsCtx *tls, unsigned char *buf, size_t len,
 	return r;
 }
 void
-ttls_write_change_cipher_spec(TlsCtx *tls)
-{
-	TlsIOCtx *io = &tls->io_out;
+ttls_write_change_cipher_spec(TlsCtx *tls, struct sg_table *sgt,
+			      unsigned char **in_buf)
+{
+	/* The ChangeCipherSpec message is added after NewSessionTicket one. */
+	if (unlikely(in_buf)) {
```

Comment:
Why `unlikely()`? I think the most TLS handshakes support the session tickets extension.

Note, that I added 2 more test cases into #1054. Please make sure that this branch is properly covered with the tests.

Reply:
Unlikely was replaced with likely - with tickets enabled (mostly true) ChangeCipherSpec message will always be the second message in the sequence. Comment was updated.